src/gui/symbology-ng/qgsrendererv2propertiesdialog.cpp

```diff
@@ -222,7 +222,6 @@ void QgsRendererV2PropertiesDialog::rendererChanged()
     {
       if ( mMapCanvas )
         mActiveWidget->setMapCanvas( mMapCanvas );
-      changeOrderBy( mActiveWidget->renderer()->orderBy(), mActiveWidget->renderer()->orderByEnabled() );
```

Comment:
The way in which this has been implemented is by no means perfect. Silently overwriting different settings (layer variables, order by, style...) by just switching the combobox is not the best UX ever. So if you have a good suggestion how this could be improved, feel free to tell :)